src/main/scala/is/hail/io/gen/GenLoader.scala

```diff
@@ -124,11 +75,7 @@ object GenLoader {
       val d1 = gp(i + 1)
       val d2 = gp(i + 2)
       val sumDosages = d0 + d1 + d2
-      if (sumDosages == 0.0)
-        reportAcc += GenReport.gpNoCall
-      else if (math.abs(sumDosages - 1.0) > tolerance)
-        reportAcc += GenReport.gpSumLessThanTolerance
-      else {
+      if (sumDosages != 0.0 && math.abs(sumDosages - 1.0) <= tolerance) {
```

Comment:
You don't need the `sumDosages != 0` check. This will be caught by the tolerance test.